src/prefect/engine/task_runner.py

```diff
@@ -168,7 +168,7 @@ def initialize_run(  # type: ignore
         context.update(
             task_run_count=run_count,
             task_name=self.task.name,
-            task_tags=self.task.tags,
+            task_tags=list(self.task.tags),
```

Comment:
Nitpick but I think users will expect this to be a set because that's how it's defined on the task object. Since you have the `list()` conversion in the state class itself, I think it's ok to remove it here.
```suggestion
            task_tags=self.task.tags,
```

Reply:
Will need to update some tests real quick

Reply:
Updated.